chainer/functions/evaluation/binary_accuracy.py

```diff
@@ -27,7 +29,7 @@ def forward(self, inputs):
         t = t.ravel()
         c = (y >= 0)
         count = xp.maximum(1, (t != self.ignore_label).sum())
-        return xp.asarray((c == t).sum(dtype='f') / count, dtype='f'),
+        return xp.asarray((c == t).sum() / count, dtype=y.dtype),
```

Comment:
It uses Python3 mode.
https://github.com/pfnet/chainer/pull/1165/files/91b37a010ae0ff64115e1fbc54b427f292b9920c#diff-36d8cd578e4268189b044f7fad32c5e3R1